frontend/src/views/producer/dashboard/edition-form/ButtonBar.jsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import styled, { withTheme } from 'styled-components';
+import AntButton from 'antd/lib/button';
+
+const Wrapper = styled.div`
+  margin-bottom: 1rem;
+`;
+
+const Button = styled(AntButton)`
+  && {
+    border-radius: 8px;
+    color: ${props => props.bgColor};
+    border-color: transparent;
+
+    &:hover {
+      color: white;
+      background-color: ${props => props.bgColor};
+      box-shadow: 0 3px 6px rgba(0, 0, 0, 0.15);
+    }
+    ${props =>
+      props.disabled && 'filter: grayscale(100%); &:hover {box-shadow: none};'};
+  }
+`;
+
+class ButtonBar extends React.Component {
+  render() {
+    const { palette } = this.props.theme;
+    const {
+      addNewEventCB,
+      confirmChangeCB,
+      discardChangeCB,
+      canConfirm,
+      canAddNew,
+    } = this.props;
+    return (
+      <Wrapper>
+        <AntButton.Group>
+          <Button
+            disabled={!canAddNew}
+            bgColor={palette.primaryDeep}
+            icon="plus"
+            onClick={addNewEventCB}
+            onTouchEnd={addNewEventCB}
+          >
+            Add new event
+          </Button>
+          <Button
+            disabled={!canConfirm}
+            bgColor={palette.deepGreen}
+            icon="check"
+            onClick={confirmChangeCB}
+            onTouchEnd={confirmChangeCB}
+          >
+            Confirm change
+          </Button>
+          <Button
+            disabled={!canConfirm}
+            bgColor={palette.red}
+            icon="close"
+            onClick={discardChangeCB}
+            onTouchEnd={discardChangeCB}
```

Comment:
Are you sure the action doesn't get fired twice in mobile, since I guess it is registering onClick as well as onTouchEnd in mobile devices?

Reply:
im not so sure, but since I have just been doing producerUI, which is meant to be used for desktop, i cant know
I can fix this once I notice any thing bad in the consumer UI